src/components/melding/NyMelding.tsx

```diff
@@ -0,0 +1,294 @@
+import { Alert, Box, Button, ErrorMessage, Heading, HStack, Textarea, VStack } from '@navikt/ds-react';
+import { EnvelopeClosedIcon } from '@navikt/aksel-icons';
+import { ReactElement } from 'react';
+import { Temagruppe } from 'src/models/temagrupper';
+import VelgTema from 'src/components/melding/VelgTema';
+import { MeldingsType, meldingsTyperTekst, VelgMeldingsType } from 'src/components/melding/VelgMeldingsType';
+import VelgOppgaveliste, { Oppgaveliste } from 'src/components/melding/VelgOppgaveliste';
+import { ValgForMeldingstype } from 'src/components/melding/ValgForMeldingstype';
+import VelgSak from 'src/components/melding/VelgSak';
+import AvsluttDialogEtterSending from 'src/components/melding/AvsluttDialogEtterSending';
+import { FieldApi, useForm, useStore } from '@tanstack/react-form';
+import { z } from 'zod';
+import {
+    JournalforingsSak
+} from 'src/app/personside/infotabs/meldinger/traadvisning/verktoylinje/journalforing/JournalforingPanel';
+import { useValgtenhet } from 'src/context/valgtenhet-state';
+import saksbehandlersEnheter from 'src/rest/resources/saksbehandlersEnheterResource';
+import persondataResource from 'src/rest/resources/persondataResource';
+import { capitalizeName } from 'src/utils/string-utils';
+import { useFodselsnummer } from 'src/utils/customHooks';
+import { $api } from 'src/lib/clients/modiapersonoversikt-api';
+import { type SendMeldingRequestV2, SendMeldingRequestV2TraadType } from 'src/generated/modiapersonoversikt-api';
+import { Link } from '@tanstack/react-router';
+
+interface NyMeldingProps {
+    lukkeKnapp?: ReactElement<typeof Button>;
+}
+
+function NyMelding({ lukkeKnapp }: NyMeldingProps) {
+    const fnr = useFodselsnummer();
+    const enhetsId = useValgtenhet().enhetId;
+    const enhetsNavn = useEnhetsnavn(enhetsId);
+    const brukerNavn = useBrukernavn();
+
+    const defaultFormOptions: NyMeldingFormOptions = {
+        meldingsType: MeldingsType.Referat,
+        melding: '',
+        tema: undefined,
+        oppgaveliste: Oppgaveliste.MinListe,
+        sak: undefined,
+        fnr: fnr ?? '',
+        enhetsId: enhetsId ?? ''
+    };
+    const { error, mutate, isPending, isSuccess } = $api.useMutation('post', '/rest/v2/dialog/sendmelding', {
+        onSuccess: () => {
+            form.reset({
+                ...defaultFormOptions,
+                meldingsType: form.state.values.meldingsType
+            }, { keepDefaultValues: true });
+        }
+    });
+
+    const schema = nyMeldingSchema();
+    const form = useForm<NyMeldingFormOptions>({
```

Comment:
```suggestion
    const form = useForm({
```

Dette inferes av formet fra schemaet i `validators` så du trenger verken type parameteren eller typen selv.